entrance_rando.py

```diff
@@ -0,0 +1,388 @@
+    # similar to fill, skip validity checks on entrances if the game is beatable on minimal accessibility
+    perform_validity_check = True
+
+    def do_placement(source_exit: Entrance, target_entrance: Entrance) -> None:
+        placed_exits, removed_entrances = er_state.connect(source_exit, target_entrance)
+        # remove the placed targets from consideration
+        for entrance in removed_entrances:
+            entrance_lookup.remove(entrance)
+        # propagate new connections
+        er_state.collection_state.update_reachable_regions(world.player)
+        if on_connect:
+            on_connect(er_state, placed_exits)
+
+    def find_pairing(dead_end: bool, require_new_regions: bool) -> bool:
+        nonlocal perform_validity_check
+        placeable_exits = er_state.find_placeable_exits(perform_validity_check)
+        for source_exit in placeable_exits:
+            target_groups = target_group_lookup[source_exit.randomization_group]
+            for target_entrance in entrance_lookup.get_targets(target_groups, dead_end, preserve_group_order):
+                # requiring a new region is a proxy for enforcing new entrances are added, thus growing the search
+                # space. this is not quite a full fidelity conversion, but doesn't seem to cause issues enough
+                # to do more complex checks.
+                # the new region requirement can be ignored on a beaten minimal, islands are no issue there
+                region_requirement_satisfied = (not perform_validity_check or not require_new_regions
+                                                or target_entrance.connected_region not in er_state.placed_regions)
+                if region_requirement_satisfied and source_exit.can_connect_to(target_entrance, er_state):
+                    do_placement(source_exit, target_entrance)
+                    return True
+        else:
+            # no source exits had any valid target so this stage is deadlocked. retries may be implemented if early
+            # deadlocking is a frequent issue.
+            lookup = entrance_lookup.dead_ends if dead_end else entrance_lookup.others
+
+            # if we're in a stage where we're trying to get to new regions, we could also enter this
+            # branch in a success state (when all regions of the preferred type have been placed, but there are still
+            # additional unplaced entrances into those regions)
+            if require_new_regions:
+                if all(e.connected_region in er_state.placed_regions for e in lookup):
+                    return False
+
+            # if we're on minimal accessibility and can guarantee the game is beatable,
+            # we can prevent a failure by bypassing future validity checks. this check may be
+            # expensive; fortunately we only have to do it once
+            if perform_validity_check and world.options.accessibility == Accessibility.option_minimal \
```

Comment:
We discussed this on the PR, but this indicates that `minimal` still tries to make a complete map, and only "prevents a failure state" that would normally crash.

As a player, I would expect that `minimal` + ER would lead to half my regions being inaccessible most of the time.

However, as I said in that discussion, I also don't think there's a super elegant solution to this. We came up with some things that could be tried, but for me, they all had shortcomings.

Either way, it's something I find very much "noteworthy"

Reply:
Updated docs.